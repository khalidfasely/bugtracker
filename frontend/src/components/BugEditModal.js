import Modal from "react-modal";

import NewBug from "./NewBug";

const BugEditModal = ({
    editModalOpen, setEditModalOpen,
    users, bug
}) => (
    <Modal
        isOpen={editModalOpen}
        onRequestClose={() => setEditModalOpen(false)}
    >
        <NewBug users={users} isEdit={true} bug={bug} setEditModalOpen={setEditModalOpen} />
    </Modal>
);

export default BugEditModal;