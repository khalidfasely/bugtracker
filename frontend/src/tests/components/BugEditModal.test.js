import React from "react";
import { render, fireEvent } from "@testing-library/react";

import BugEditModal from "../../components/BugEditModal";

jest.mock('../../components/NewBug', () => () => <div>NewBug</div>);

let props;

beforeEach(() => {//editModalOpen, setEditModalOpen,
    props = {
        editModalOpen: true,
        setEditModalOpen: jest.fn()
    };
});

test('Should render BugEditModal component correctly', () => {
    const { asFragment } = render(<BugEditModal {...props} />);
    expect(asFragment()).toMatchSnapshot();
});