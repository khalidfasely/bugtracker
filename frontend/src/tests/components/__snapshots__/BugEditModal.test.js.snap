// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BugEditModal component correctly 1`] = `
<DocumentFragment>
  <div>
    NewBug
  </div>
</DocumentFragment>
`;
